### PR TITLE
SI-7240 fixes language feature lookup

### PR DESCRIPTION
**Pull request by xeno-by**
As I discovered today, Definitions.getMember have a fallback clause,
which accounts for the phases which have inner classes flattened.

This fallback uses nme.flattenedName to compute a flattened name, but
unfortunately nme.flattenedName produces a TermName, not a TypeName,
which means that the fallback will commence search in a wrong namespace
with predictable results.

### CODE REVIEW

**Comment by xeno-by** [comment]
review @retronym @vjovanov

File: src/reflect/scala/reflect/internal/Definitions.scala [hunk 1]
```diff
@@ -1046,7 +1046,7 @@ trait Definitions extends api.StandardDefinitions {
       getMemberIfDefined(owner, name) orElse {
         if (phase.flatClasses && name.isTypeName && !owner.isPackageObjectOrClass) {
           val pkg = owner.owner
-          val flatname = nme.flattenedName(owner.name, name)
+          val flatname = tpnme.flattenedName(owner.name, name)
```
Comment (retronym) [inline src/reflect/scala/reflect/internal/Definitions.scala:1049]: Elsewhere:

``` scala
class ClassSymbol {
...
    override def name: TypeName = {
      if (Statistics.canEnable) Statistics.incCounter(nameCount)
      if (needsFlatClasses) {
        if (flatname eq null)
          flatname = nme.flattenedName(rawowner.name, rawname).toTypeName

        flatname
      }
      else rawname
    }
```

I don't have a strong opinion as to which is clearer, nor an idea which is faster, but perhaps we can keep the two consistent.

Comment (xeno-by) [inline src/reflect/scala/reflect/internal/Definitions.scala:1049]: Done

**Comment by vjovanov** [comment]
LGTM. LMS, with all the Scala packages mixed in, now works. Thanks.

**Comment by retronym** [comment]
LGTM

**Pull request merged**